python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -751,6 +895,117 @@ def from_rule_evaluation_results(
+
+    def get_rule_evaluation_results(
+        self, rule_snapshot: AutoMaterializeRuleSnapshot, asset_graph: AssetGraph
+    ) -> RuleEvaluationResults:
+        """For a given rule snapshot, returns the calculated evaluations for that rule."""
+        results = []
+        for serialized_result in self.partition_subsets_by_condition:
+            # filter for the same rule
+            if serialized_result[0].rule_snapshot != rule_snapshot:
+                continue
+            deserialized_result = self._deserialize_result(serialized_result, asset_graph)
+            if deserialized_result:
+                results.append(deserialized_result)
+        return results
+
+    def _get_asset_partitions_with_decision_type(
+        self, decision_type: AutoMaterializeDecisionType, asset_graph: AssetGraph
+    ) -> AbstractSet[AssetKeyPartitionKey]:
+        """Returns the set of asset partitions with a given decision type applied to them."""
+        asset_partitions = set()
+        for serialized_result in self.partition_subsets_by_condition:
+            if serialized_result[0].rule_snapshot.decision_type != decision_type:
+                continue
+            deserialized_result = self._deserialize_result(serialized_result, asset_graph)
+            if deserialized_result is None:
+                continue
+            asset_partitions.update(deserialized_result[1])
+        return asset_partitions
+
+    def get_requested_or_discarded_asset_partitions(
+        self, asset_graph: AssetGraph
+    ) -> AbstractSet[AssetKeyPartitionKey]:
+        """Returns the set of asset partitions which were either requested or discarded on this
+        evaluation.
+        """
+        to_materialize = self._get_asset_partitions_with_decision_type(
+            AutoMaterializeDecisionType.MATERIALIZE, asset_graph
+        )
+        if not to_materialize:
+            return set()
+        to_skip = self._get_asset_partitions_with_decision_type(
+            AutoMaterializeDecisionType.SKIP, asset_graph
+        )
+        return to_materialize - to_skip
+
+    def get_evaluated_asset_partitions(
+        self, asset_graph: AssetGraph
+    ) -> AbstractSet[AssetKeyPartitionKey]:
+        """Returns the set of asset partitions which were evaluated by any rule on this evaluation."""
+        # no asset partition can be evaluated by SKIP or DISCARD rules without having at least one
+        # materialize rule evaluation
+        return self._get_asset_partitions_with_decision_type(
+            AutoMaterializeDecisionType.MATERIALIZE, asset_graph
+        )
+
+    def equivalent_to_stored_evaluation(
+        self, stored_evaluation: Optional["AutoMaterializeAssetEvaluation"], asset_graph: AssetGraph
+    ) -> bool:
+        """This function returns if a stored record is equivalent to this one. To do so, we can't
+        just use regular namedtuple equality, as the serialized partition subsets will be
+        potentially have different string values.
```

Comment:
What's a situation where they would have different string values? Is it due to the undeterministic ordering of partitions? I wonder if we can just make this more deterministic.

Reply:
Yep this is purely due to undeterministic string orderings. I can make them deterministic in a followup and remove this line there